File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/dto/request/CreateSubmissionRequestDTO.kt
package io.leonfoliveira.judge.api.dto.request

import io.leonfoliveira.judge.core.domain.enumerate.Language
import io.leonfoliveira.judge.core.domain.model.RawAttachment
import io.leonfoliveira.judge.core.service.dto.input.CreateSubmissionInputDTO
import java.util.Base64

data class CreateSubmissionRequestDTO(
    val language: Language,
    val code: AttachmentRequestDTO,
) {
    fun toInputDTO(): CreateSubmissionInputDTO {
        return CreateSubmissionInputDTO(
            language = language,
            code =
                RawAttachment(
                    filename = code.filename,
                    content = Base64.getDecoder().decode(code.content),
                ),
        )
    }
}

File: applications/webapp/src/core/port/dto/request/CreateSubmissionRequestDTO.ts
import { SubmissionLanguage } from "@/core/domain/enumerate/SubmissionLanguage";
import { AttachmentResponseDTO } from "@/core/port/dto/response/attachment/AttachmentResponseDTO";

export type CreateSubmissionRequestDTO = {
  problemId: string;
  language: SubmissionLanguage;
  code: AttachmentResponseDTO;
};
